src/main/java/com/nutomic/syncthingandroid/syncthing/RestApi.java

```diff
@@ -204,6 +204,10 @@ public void onWebGuiAvailable() {
         new GetTask() {
             @Override
             protected void onPostExecute(String s) {
+                if (s == null) {
+                    Log.w(TAG, "Empty JSON string when retrieving version information");
```

Comment:
So I'd say get rid of the changes in this file, and just change GetTask (which I haven't reviewed yet).